File: Tokens/stockpaydividend/contracts/src/test/java/net/corda/samples/stockpaydividend/contracts/ContractTests.java
package net.corda.samples.stockpaydividend.contracts;

import net.corda.core.contracts.UniqueIdentifier;
import net.corda.core.identity.CordaX500Name;
import net.corda.samples.stockpaydividend.states.StockState;
import net.corda.testing.core.TestIdentity;
import net.corda.testing.node.MockServices;
import org.junit.Test;

import java.math.BigDecimal;
import java.util.Date;

import static net.corda.testing.node.NodeTestUtils.ledger;

public class ContractTests {
    private final MockServices ledgerServices = new MockServices();
    TestIdentity Operator = new TestIdentity(new CordaX500Name("Alice",  "TestLand",  "US"));

    //sample tests
    @Test
    public void multipleOutputTests() {
        StockState tokenPass = new StockState(new UniqueIdentifier(),Operator.getParty(),"TT","Test Token",
                "USD",BigDecimal.valueOf(2.7), BigDecimal.valueOf(0.2),new Date(),new Date());
        StockState tokenFail = new StockState(new UniqueIdentifier(),Operator.getParty(),"","Test Token",
                "USD",BigDecimal.valueOf(2.7), BigDecimal.valueOf(0.2),new Date(),new Date());
        ledger(ledgerServices, l -> {
            l.transaction(tx -> {
                tx.output(StockContract.CONTRACT_ID, tokenFail);
                tx.command(Operator.getPublicKey(), new com.r3.corda.lib.tokens.contracts.commands.Create()); //Fail because multiple outputs
                return tx.fails();
            });
            l.transaction(tx -> {
                tx.output(StockContract.CONTRACT_ID, tokenPass);
                tx.command(Operator.getPublicKey(), new com.r3.corda.lib.tokens.contracts.commands.Create());
                return tx.verifies();
            });
            return null;
        });
    }
}

File: kotlin-samples/Tokens/stockpaydividend/contracts/src/test/kotlin/net/corda/samples/stockpaydividend/contracts/ContractTests.kt
package net.corda.samples.stockpaydividend.contracts

import net.corda.core.contracts.UniqueIdentifier
import net.corda.core.identity.CordaX500Name
import net.corda.samples.stockpaydividend.states.StockState
import net.corda.testing.core.TestIdentity
import net.corda.testing.node.MockServices
import net.corda.testing.node.ledger
import org.junit.Test
import java.math.BigDecimal
import java.util.*

class ContractTests {
    private val ledgerServices = MockServices()
    val Operator = TestIdentity(CordaX500Name(organisation = "Alice", locality = "TestLand", country = "US"))

    //sample tests
    @Test
    fun `multiple Output Tests`() {
        val tokenPass = StockState(Operator.party, "TT", "Test Token",
                "USD", BigDecimal.valueOf(2.7), BigDecimal.valueOf(0), Date(), Date(), UniqueIdentifier()
        )
        val tokenFail = StockState(Operator.party, "", "Test Token",
                "USD", BigDecimal.valueOf(2.7), BigDecimal.valueOf(0.7), Date(), Date(), UniqueIdentifier()
        )
        ledgerServices.ledger {
            transaction {
                output(StockContract.CONTRACT_ID, tokenFail)
                output(StockContract.CONTRACT_ID, tokenFail)
                command(Operator.publicKey, com.r3.corda.lib.tokens.contracts.commands.Create())
                this.fails()
            }
        }
        ledgerServices.ledger {
            transaction {
                output(StockContract.CONTRACT_ID, tokenPass)
                command(Operator.publicKey, com.r3.corda.lib.tokens.contracts.commands.Create())
                this.verifies()
            }
        }
    }
}